Cannot build on macOS using doc instructions

I'm following [these](https://clickhouse.com/docs/en/development/build-osx) steps in the docs to build on macOS. Perhaps there is a step i'm missing? I'm using the recommended commands

```
brew update
brew install ccache cmake ninja libtool gettext llvm gcc binutils grep findutils nasm
```
```
git clone --recursive git@github.com:ClickHouse/ClickHouse.git
```
```
cd ClickHouse
mkdir build
export PATH=$(brew --prefix llvm)/bin:$PATH
cmake -G Ninja -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_C_COMPILER=$(brew --prefix llvm)/bin/clang -DCMAKE_CXX_COMPILER=$(brew --prefix llvm)/bin/clang++ -S . -B build
cmake --build build
``` 
but after running `cmake --build build` I get this error
```
FAILED: src/CMakeFiles/clickhouse_common_io.dir/Common/waitForPid.cpp.o 
/usr/local/bin/ccache /usr/local/opt/llvm/bin/clang++ -DBOOST_ASIO_HAS_STD_INVOKE_RESULT=1 -DBOOST_ASIO_STANDALONE=1 -DBOOST_TIMER_ENABLE_DEPRECATED=1 -DCARES_STATICLIB -DDUMMY_BACKTRACE -DENABLE_MULTITARGET_CODE=1 -DHAVE_BZLIB_H=1 -DHAVE_CONFIG_H -DHAVE_LIBLZMA=1 -DHAVE_LIBZSTD=1 -DHAVE_LIBZSTD_COMPRESSOR=1 -DHAVE_LZMA_H=1 -DHAVE_ZLIB_H=1 -DHAVE_ZSTD_H=1 -DINCBIN_SILENCE_BITCODE_WARNING -DLZ4_DISABLE_DEPRECATE_WARNINGS=1 -DLZ4_FAST_DEC_LOOP=1 -DMZ_COMPAT_VERSION=110 -DPOCO_ENABLE_CPP11 -DPOCO_HAVE_FD_POLL -DPOCO_OS_FAMILY_UNIX -DSTD_EXCEPTION_HAS_STACK_TRACE=1 -DUNALIGNED_OK -DWITH_COVERAGE=0 -DWITH_GZFILEOP -DX86_64 -DZLIB_COMPAT -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS -I/Users/alexholland/Projects/ClickHouse/build/includes/configs -I/Users/alexholland/Projects/ClickHouse/src -I/Users/alexholland/Projects/ClickHouse/build/src -I/Users/alexholland/Projects/ClickHouse/build/src/Core/include -I/Users/alexholland/Projects/ClickHouse/base/widechar_width -I/Users/alexholland/Projects/ClickHouse/base/base/.. -I/Users/alexholland/Projects/ClickHouse/build/base/base/.. -I/Users/alexholland/Projects/ClickHouse/contrib/cctz/include -I/Users/alexholland/Projects/ClickHouse/contrib/re2 -I/Users/alexholland/Projects/ClickHouse/base/pcg-random/. -I/Users/alexholland/Projects/ClickHouse/contrib/libfiu/libfiu -I/Users/alexholland/Projects/ClickHouse/contrib/libssh/include -I/Users/alexholland/Projects/ClickHouse/build/contrib/libssh/include -I/Users/alexholland/Projects/ClickHouse/contrib/miniselect/include -I/Users/alexholland/Projects/ClickHouse/contrib/zstd/lib -I/Users/alexholland/Projects/ClickHouse/contrib/pocketfft -I/Users/alexholland/Projects/ClickHouse/contrib/lz4/lib -I/Users/alexholland/Projects/ClickHouse/contrib/libarchive-cmake -I/Users/alexholland/Projects/ClickHouse/contrib/libarchive/libarchive -I/Users/alexholland/Projects/ClickHouse/contrib/minizip-ng -I/Users/alexholland/Projects/ClickHouse/contrib/minizip-ng-cmake -isystem /Users/alexholland/Projects/ClickHouse/contrib/llvm-project/libcxx/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/llvm-project/libcxxabi/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/libunwind/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/jemalloc-cmake/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/sparsehash-c11 -isystem /Users/alexholland/Projects/ClickHouse/contrib/incbin -isystem /Users/alexholland/Projects/ClickHouse/contrib/cityhash102/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/boost -isystem /Users/alexholland/Projects/ClickHouse/base/poco/Net/include -isystem /Users/alexholland/Projects/ClickHouse/base/poco/Foundation/include -isystem /Users/alexholland/Projects/ClickHouse/base/poco/NetSSL_OpenSSL/include -isystem /Users/alexholland/Projects/ClickHouse/base/poco/Crypto/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/openssl-cmake/darwin_x86_64/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/openssl/include -isystem /Users/alexholland/Projects/ClickHouse/base/poco/Util/include -isystem /Users/alexholland/Projects/ClickHouse/base/poco/JSON/include -isystem /Users/alexholland/Projects/ClickHouse/base/poco/XML/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/replxx/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/fmtlib-cmake/../fmtlib/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/magic_enum/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/double-conversion -isystem /Users/alexholland/Projects/ClickHouse/contrib/dragonbox/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/libdivide-cmake/. -isystem /Users/alexholland/Projects/ClickHouse/contrib/libdivide -isystem /Users/alexholland/Projects/ClickHouse/contrib/zlib-ng -isystem /Users/alexholland/Projects/ClickHouse/build/contrib/zlib-ng-cmake -isystem /Users/alexholland/Projects/ClickHouse/contrib/libcpuid -isystem /Users/alexholland/Projects/ClickHouse/contrib/abseil-cpp -isystem /Users/alexholland/Projects/ClickHouse/contrib/pdqsort -isystem /Users/alexholland/Projects/ClickHouse/contrib/xz/src/liblzma/api -isystem /Users/alexholland/Projects/ClickHouse/contrib/c-ares/src/lib -isystem /Users/alexholland/Projects/ClickHouse/contrib/c-ares/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/c-ares-cmake/darwin -isystem /Users/alexholland/Projects/ClickHouse/contrib/aws/src/aws-cpp-sdk-core/include -isystem /Users/alexholland/Projects/ClickHouse/build/contrib/aws-cmake/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/aws/generated/src/aws-cpp-sdk-s3/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/aws-c-auth/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/aws-c-common/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/aws-c-io/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/aws-crt-cpp/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/aws-c-mqtt/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/aws-c-sdkutils/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/azure/sdk/core/azure-core/inc -isystem /Users/alexholland/Projects/ClickHouse/contrib/azure/sdk/identity/azure-identity/inc -isystem /Users/alexholland/Projects/ClickHouse/contrib/azure/sdk/storage/azure-storage-common/inc -isystem /Users/alexholland/Projects/ClickHouse/contrib/azure/sdk/storage/azure-storage-blobs/inc -isystem /Users/alexholland/Projects/ClickHouse/contrib/brotli/c/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/snappy -isystem /Users/alexholland/Projects/ClickHouse/build/contrib/snappy-cmake -isystem /Users/alexholland/Projects/ClickHouse/contrib/libbcrypt -isystem /Users/alexholland/Projects/ClickHouse/contrib/msgpack-c/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/fast_float/include -isystem /Users/alexholland/Projects/ClickHouse/contrib/bzip2 -isystem /Users/alexholland/Projects/ClickHouse/build/contrib/bzip2 -fdiagnostics-color=always -Xclang -fuse-ctor-homing -Wno-enum-constexpr-conversion -fsized-deallocation  -gdwarf-aranges -pipe -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -ffile-prefix-map=/Users/alexholland/Projects/ClickHouse=. -falign-functions=32 -mbranches-within-32B-boundaries -ffp-contract=off  -stdlib=libc++ -fdiagnostics-absolute-paths -fstrict-vtable-pointers -Wall -Wextra -Wframe-larger-than=65536 -Weverything -Wpedantic -Wno-zero-length-array -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c++20-compat -Wno-sign-conversion -Wno-implicit-int-conversion -Wno-implicit-int-float-conversion -Wno-ctad-maybe-unsupported -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-padded -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -Wno-thread-safety-negative -Wno-enum-constexpr-conversion -Wno-unsafe-buffer-usage -Wno-switch-default -O2 -g -DNDEBUG -O3 -g  -std=c++23 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk -mmacosx-version-min=10.15   -D OS_DARWIN -D _DARWIN_C_SOURCE -Werror -Wno-deprecated-declarations -Wno-poison-system-directories -nostdinc++ -MD -MT src/CMakeFiles/clickhouse_common_io.dir/Common/waitForPid.cpp.o -MF src/CMakeFiles/clickhouse_common_io.dir/Common/waitForPid.cpp.o.d -o src/CMakeFiles/clickhouse_common_io.dir/Common/waitForPid.cpp.o -c /Users/alexholland/Projects/ClickHouse/src/Common/waitForPid.cpp
/Users/alexholland/Projects/ClickHouse/src/Common/waitForPid.cpp:135:39: error: missing field 'filter' initializer [-Werror,-Wmissing-field-initializers]
  135 |     struct kevent change = {.ident = 0};
      |                                       ^
/Users/alexholland/Projects/ClickHouse/src/Common/waitForPid.cpp:147:38: error: missing field 'filter' initializer [-Werror,-Wmissing-field-initializers]
  147 |     struct kevent event = {.ident = 0};
      |                                      ^
2 errors generated.
[4122/10671] Building CXX object src/CMakeFiles/clickhouse_common_io.dir/IO/AsynchronousReadBufferFromFile.cpp.o
ninja: build stopped: subcommand failed.
```

**Operating system**

local machine: macOS Monterery, Version 12.7.4

**Cmake version**

3.29.2

**Ninja version**

1.12.0

**Compiler name and version**
```
Homebrew clang version 18.1.4
Target: x86_64-apple-darwin21.6.0
Thread model: posix
InstalledDir: /usr/local/opt/llvm/bin
```

--- a/src/Common/waitForPid.cpp
+++ b/src/Common/waitForPid.cpp
@@ -132,7 +132,9 @@ static PollPidResult pollPid(pid_t pid, int timeout_in_ms)
     if (kq == -1)
         return PollPidResult::FAILED;
 
-    struct kevent change = {.ident = 0};
+    struct kevent change;
+    change.ident = 0;
+
     EV_SET(&change, pid, EVFILT_PROC, EV_ADD, NOTE_EXIT, 0, NULL);
 
     int event_add_result = HANDLE_EINTR(kevent(kq, &change, 1, NULL, 0, NULL));
@@ -144,7 +146,9 @@ static PollPidResult pollPid(pid_t pid, int timeout_in_ms)
         return PollPidResult::FAILED;
     }
 
-    struct kevent event = {.ident = 0};
+    struct kevent event;
+    event.ident = 0;
+
     struct timespec remaining_timespec = {.tv_sec = timeout_in_ms / 1000, .tv_nsec = (timeout_in_ms % 1000) * 1000000};
     int ready = HANDLE_EINTR(kevent(kq, nullptr, 0, &event, 1, &remaining_timespec));
     PollPidResult result = ready < 0 ? PollPidResult::FAILED : PollPidResult::RESTART;

